tsdb/head.go

```diff
@@ -1558,6 +1603,7 @@ func (s *stripeSeries) gc(mint int64) (map[uint64]struct{}, int) {
 			for _, series := range all {
 				series.Lock()
 				rmChunks += series.truncateChunksBefore(mint)
+				series.cleanupExtraWriteIds()
```

Comment:
This doesn't feel right. Shouldn't we block GC of a series if it'd break isolation?

Reply:
@gouthamve did you have any thoughts about this?

Reply:
I will remove the cleanup of extra write IDs as it doesn't seem to be necessary and also seems to be buggy.